### PR TITLE
[Transform] Revamp transform config and query validation code

### LINKED ISSUE #59588: [Transform] use ValidationExceptions where possible
followup from #58744

validateConfig/Query should use ValidationException, probably some other parts of the code, too.

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR improves a few aspects of validation in transforms codebase:
- Replaces usage of `ElasticsearchStatusException` with `ValidatonException` where possible
- Removes `isValid` method and uses `validate` method instead
- Removes unreachable code of handling validation exceptions in actions
- Renames `getAdditionalValidations` method to `getAdditionalSourceDestValidations` to distinguish these from other validations

Closes https://github.com/elastic/elasticsearch/issues/59588

### CODE REVIEW

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformPersistentTasksExecutor.java [hunk 1]
```diff
@@ -280,12 +281,18 @@ protected void nodeOperation(AllocatedPersistentTask task, @Nullable TransformTa
...
         ActionListener<TransformConfig> getTransformConfigListener = ActionListener.wrap(config -> {
-            if (config.isValid()) {
+            ValidationException validationException = config.validate(null);
+            if (validationException == null) {
                 indexerBuilder.setTransformConfig(config);
                 SchemaUtil.getDestinationFieldMappings(buildTask.getParentTaskClient(), config.getDestination().getIndex(),
                         getFieldMappingsListener);
             } else {
-                markAsFailed(buildTask, TransformMessages.getMessage(TransformMessages.TRANSFORM_CONFIGURATION_INVALID, transformId));
+                markAsFailed(
```
Comment (hendrikmuhs) [inline x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformPersistentTasksExecutor.java:290]: if I remember correctly `markAsFailed` does not audit. I think we should add an audit here, so users will see the error in the job messages.

**Review by hendrikmuhs — COMMENTED** [review]
Nice! Much simpler.

I think we should have proper error messages as the errors will be written into the reason when the task fails to start.

Could you also add some tests for the messages added?

**Review by przemekwitek — COMMENTED** [review]
> I think we should have proper error messages as the errors will be written into the reason when the task fails to start.

+1

> Could you also add some tests for the messages added?

Added test assertions in unit tests. Are there other places that you'd find more tests beneficial?

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformPersistentTasksExecutor.java (hunk 1 above)
Comment (hendrikmuhs) [reply x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformPersistentTasksExecutor.java:291]: you are right, this looks inconsistent, but its already inconsistent today. The last step audits, too. The in-between ones don't audit but only log.
I think we should audit in all failure cases, logs might not be accessible to the user.

**Review by hendrikmuhs — APPROVED** [review]
LGTM

**Pull request merged**